core/src/main/java/org/opensearch/sql/expression/operator/convert/TypeCastOperators.java

```diff
@@ -173,6 +176,13 @@ private static DefaultFunctionResolver castToBoolean() {
         impl(nullMissingHandling((v) -> v), BOOLEAN, BOOLEAN));
   }
 
+  private static DefaultFunctionResolver castToIp() {
+    return FunctionDSL.define(
+        BuiltinFunctionName.CAST_TO_IP.getName(),
+        impl(nullMissingHandling((v) -> v), IP, IP),
```

Comment:
ordering matters - make sure to but the naive cast last

Reply:
Interesting. What is the potential effect of having them in the "wrong" order?

Reply:
There may be an implicit casting on an earlier operation.